CommonMC/fcl/prolog.fcl

```diff
@@ -30,7 +32,7 @@ CommonMC: {
 # Event window marker
     EWMProducer : { module_type : EventWindowMarkerProducer }
   }
-  TimeMaps : [ protonTimeMap, muonTimeMap, cosmicTimeMap ]
+  TimeMaps        : @local::TIME_MAPS
```

Comment:
Hi Pasha,
All this seems to do is change the name TimeMaps ->TIME_MAPS and move the definition outside the namespace.  We already use CommonMC.TimeMaps in many places, I don't see how creating this alternate name makes the use more consistent.

Reply:
let me explain: 

the proposed change has to do not with renaming, but exactly with the intention to continue use CommonMC.TimeMaps, while being able to redefine it's value consistently in a complex FCL configuration with multiple levels of nesting. Unfortunately, currently is it not possible without editing the CommonMC/fcl/prolog.fcl.

You can see what the proposed change does by running art (mu2e) with --debug-config switch 
with the two fcl files below and compare the expanded output.

Please let me know, if there is a better way to redefine the value of CommonMC.TimeMaps.

----------------------------------- a1.fcl
BEGIN_PROLOG
  CommonMC.TimeMaps @protect_ignore: [a, b , c]
END_PROLOG

BEGIN_PROLOG
CommonMC: {
  TimeMaps     : [a, b]
}
END_PROLOG

a : @local::CommonMC.TimeMaps
--------------------------------------------------- a2.fcl
BEGIN_PROLOG
  TIME_MAPS @protect_ignore: [a, b , c]
END_PROLOG

BEGIN_PROLOG
  TIME_MAPS : [a, b]
  CommonMC: {
    TimeMaps     : @local::TIME_MAPS
  }
END_PROLOG

a : @local::CommonMC.TimeMaps
-----------------------------------------------------

Reply:
Hi, in the first example the secon prolog redefines the whole CommonMC, not just CommonMC.TimeMaps, this is why the protect_ignore does nothing - it is only effective at the nested level.  The second example brings protect and redefinition attempts to the same level of nesting.   
I am not a fan of using protect_ingnore in "hand written" job configs (it was requested in the first place to support grid scripts), and I think is should not be allowed in any fcl files in the repository. However this patch does not violate that requirement,  and I see how this is a practical solution to kludging a "complex configuration".   I am fine with merging this.
Thinking about other ways:  FHICL expands the RHS on assignment, like the ':=' variables in gmake.   If we also had the '=' like variables that are expanded only at the point of use, it would be possible to define a complex configuration first, then redefine a part of it that is used  in multiple places with a single assignment.